### PR TITLE
[openlayers] Fix no-redundant-jsdoc lint

### PR DESCRIPTION
**Pull request by osechet**
Continuing @iamthechad 's work on cleaning lint issues: I removed all the jsdoc type definitions that were redundant with typescript defs.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [x] ~~Provide a URL to documentation or source code which provides context for the suggested changes:~~ **there is no definition change, just jsdoc has been updated**
- [ ] ~~Increase the version number in the header if appropriate.~~ **not applicable**
- [ ] ~~If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.~~ **not applicable**

### CODE REVIEW

**Review by iamthechad — CHANGES_REQUESTED** [review]
I don't think a wholesale removal of tags is the answer. For most of the `param` and `return` tags the failure message is that the type annotation is redundant. In those cases, I think the tags should be left (with the type removed) because they offer information. I agree that the redundant tags should be removed. In cases where tags are removed, care should be taken to remove all of the description associated with that tag.

File: types/openlayers/index.d.ts [hunk 1]
```diff
@@ -13,28 +13,17 @@
...
      */
     class AssertionError extends Error {
         /**
          * Error object thrown when an assertion failed. This is an ECMA-262 Error,
          * extended with a `code` property.
          * @see {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error}
-         * @constructor
-         * @extends {Error}
-         * @implements {oli.AssertionError}
-         * @param {number} code Error code.
```
Comment (iamthechad) [inline types/openlayers/index.d.ts]: Just the `{number}` could be removed here, for example.

File: types/openlayers/index.d.ts [hunk 2]
```diff
@@ -144,18 +121,14 @@ declare module ol {
...
          * to the end of the collection.
-         * @param {!Array.<T>} arr Array.
-         * @return {ol.Collection.<T>} This collection.
          * @api stable
          */
         extend(arr: T[]): ol.Collection<T>;
 
         /**
          * Iterate over each element, calling the provided callback.
-         * @param {function(this: S, T, number, Array.<T>): *} f The function to call
```
Comment (iamthechad) [inline types/openlayers/index.d.ts]: Need to be careful not to leave dangling doc when tags are removed.

Comment (osechet) [reply types/openlayers/index.d.ts]: Oups, I guess my regex was a little bit too aggressive :flushed:

File: types/openlayers/index.d.ts (hunk 1 above)
Comment (osechet) [reply types/openlayers/index.d.ts]: Yes, you are right, this information may useful. I will revert them.

**Review by iamthechad — APPROVED** [review]
Thanks for taking this on!

**Pull request merged**